### PR TITLE
Feature/selected zone

### PR DESCRIPTION
**Pull request by EliasPaulinoAndrade**
Select zones and HomeViewController.

### CODE REVIEW

File: FinalChallenge/HouseRepresentation/ViewModels/HomeViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+class HomeViewModel {
+    weak var delegate: HomeViewModelDelegate?
+    
+    private lazy var defaultFurnitures: [FurniProtocol] = {
+        let defaultFurnitures: [FurniProtocol] = [
+            ListsTool.getFurnitures()[0], ListsTool.getFurnitures()[0],
+            ListsTool.getFurnitures()[1], ListsTool.getFurnitures()[1], ListsTool.getFurnitures()[3],
+        ListsTool.getFurnitures()[3], ListsTool.getFurnitures()[4]]
```
Comment (jessicalewinter) [inline FinalChallenge/HouseRepresentation/ViewModels/HomeViewModel.swift]: Tab this line to make this line more clear to read

File: FinalChallenge/HouseRepresentation/ViewModels/InitialViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+//  Copyright © 2019 Elias Paulino. All rights reserved.
+//
+
+import UIKit
+
+protocol InitialViewModelDelegate: AnyObject {
+    func needSetViewsForEditting()
+    func needSetViewsForNormal()
+    func needSimulateTapOnFurnitureIcon()
+}
```
Comment (jessicalewinter) [inline FinalChallenge/HouseRepresentation/ViewModels/InitialViewModel.swift]: Create other file for this delegate

File: FinalChallenge/HouseRepresentation/ViewModels/InitialViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    func furniture(atIndex index: Int?, wasDraggedToPosition position: CGPoint) {
+        guard let gridPosition = self.furni?.furniParent?.gridPositionFor(position: position) else {
+            return
+        }
+         
+        eventHandler?.needMove(toPosition: gridPosition)
+    }
+    
+    func furniture(atIndex index: Int?) {
+
```
Comment (jessicalewinter) [inline FinalChallenge/HouseRepresentation/ViewModels/InitialViewModel.swift]: remove this function if you're not using it

File: FinalChallenge/HouseRepresentation/Views/HomeView.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        self.homeViewModel.sceneWasPresent()
+        
+        infoNode.touchesEnded = { [weak self] (_, _, _) in
+            guard let self = self else { return }
+            
+            self.homeViewModel.infoNodeWasTouched()
+        }
+        
+        self.homeScene.touchesBegan = { [weak self] (_, _, _) in
+
```
Comment (jessicalewinter) [inline FinalChallenge/HouseRepresentation/Views/HomeView.swift]: * remove space inside closure
* mantain consistent. add a guard let on this closure also

File: FinalChallenge/HouseRepresentation/Views/HomeView.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        self.homeScene.touchesBegan = { [weak self] (_, _, _) in
+            
+            self?.homeViewModel.homeSceneWasTouched()
+        }
+    }
+}
+
+extension HomeView: HomeViewModelDelegate {
+    func zoneIconWastTapped() {
+        print("bla")
```
Comment (jessicalewinter) [inline FinalChallenge/HouseRepresentation/Views/HomeView.swift]: remove print

File: FinalChallenge/HouseRepresentation/Views/InitialView/InitialView.swift [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    }
+}
+
+extension InitialView: InitialViewModelDelegate {
+    func needSimulateTapOnFurnitureIcon() {
+        self.menuView.simulateTapOnFirstButton()
+    }
+    
+    func needSetViewsForEditting() {
+
```
Comment (jessicalewinter) [inline FinalChallenge/HouseRepresentation/Views/InitialView/InitialView.swift]: remove space

File: FinalChallenge/HouseRepresentation/Models/HouseTiles/CornerSideTile.swift [hunk 7]
```diff
@@ -13,18 +13,22 @@ class CornerSideTile: HouseTileProtocol {
...
-        let floorTexture = SKTexture(image: UIImage(named: "realWallCornerSide")!)
+    func definition(forState state: HouseTileState) -> SKTileDefinition? {
+        let wallTexture = SKTexture(image: UIImage(named: "realWallCornerSide")!)
+        let selectedWallTexture = SKTexture(image: UIImage(named: "selectedWallCornerSide")!)
+
         let width: CGFloat = 135
-        let height = (width * floorTexture.size().height) / floorTexture.size().width
-        
-        let definition = SKTileDefinition(texture: floorTexture, size: CGSize(width: width, height: height))
+        let height = (width * wallTexture.size().height) / wallTexture.size().width
```
[the remark this answers is not in the document]
Comment (EliasPaulinoAndrade) [reply FinalChallenge/HouseRepresentation/Models/HouseTiles/CornerSideTile.swift]: This function is default from SpriteKit.

File: FinalChallenge/HouseRepresentation/Models/IsometricHouse.swift [hunk 8]
```diff
@@ -100,6 +115,26 @@ class IsometricHouse {
                 return .empty
             }
         })
+
+        self.areas = areas
+    }
+    
+    init() {
+
```
[the remark this answers is not in the document]
Comment (EliasPaulinoAndrade) [reply FinalChallenge/HouseRepresentation/Models/IsometricHouse.swift]: Its necessary due IsometricSaver data loading.

File: FinalChallenge/HouseRepresentation/ViewModels/InitialViewModel.swift (hunk 3 above)
Comment (EliasPaulinoAndrade) [reply FinalChallenge/HouseRepresentation/ViewModels/InitialViewModel.swift]: i will already use it, but I make a comment on it.

File: FinalChallenge/HouseRepresentation/Models/HouseTiles/CornerSideTile.swift (hunk 7 above)
[the remark this answers is not in the document]
Comment (jessicalewinter) [reply FinalChallenge/HouseRepresentation/Models/HouseTiles/CornerSideTile.swift]: ok then

File: FinalChallenge/HouseRepresentation/Models/IsometricHouse.swift (hunk 8 above)
[the remark this answers is not in the document]
Comment (jessicalewinter) [reply FinalChallenge/HouseRepresentation/Models/IsometricHouse.swift]: ok then

**Pull request merged**